### PR TITLE
CA-90 & CA-91 Cloudwatch log collection for HBase and Spark | CA-134 Deprecates s3 policy arns in EMR module

### PR DESCRIPTION
**Pull request by fvazquez-caylent**
* Adds example usage of cloudwatch log collection in examples/static-hbase-logs.
* Adds example usage of cloudwatch log collection in examples/static-spark-logs.

These examples will deploy the necessary resources on every instance that belongs to an EMR Cluster in order to:

Download the Cloudwatch agent.
Install the Cloudwatch agent.
Create and mount the configuration file.
Enable the Cloudwatch agent.

In this module we also:
* Deprecates s3_policy_arns in favor of additional_policy_arns.

### CODE REVIEW

File: modules/aws-emr-iam/main.tf [hunk 1]
```diff
@@ -718,11 +718,11 @@ resource "aws_iam_role" "emr_ec2_instance_profile" {
   tags                 = var.tags
 }
 
-// The IAM role policy attachment(s) that attach s3 policy ARNs to the EMR EC2 iam role
-resource "aws_iam_role_policy_attachment" "emr_ec2_s3_policies" {
-  count      = length(var.s3_policy_arns)
+//IAM role policy attachment(s) that attach additional policy ARNs to the EMR EC2 iam role
+resource "aws_iam_role_policy_attachment" "emr_ec2_additional_policies" {
+  count      = length(var.additional_policy_arns)
```
Comment (hbarros-caylent) [inline modules/aws-emr-iam/main.tf:723]: Should we be using the "effective" approach in here as well? The customer that doesn't update to use `additional_policy_arns` var would be losing the attachments in this case, right?

Comment (hbarros-caylent) [inline modules/aws-emr-iam/main.tf:723]: another case where `for_each` and `each.value` would be better fit.

**Review by hbarros-caylent — COMMENTED** [review]
Most are nitpicky comments, but do take a look and respond about the attachments in `modules/aws-emr-iam/main.tf`

File: examples/static-hbase-logs/emr.tf [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  source     = "terraform-aws-modules/key-pair/aws"
+  version    = "1.0.0"
+  key_name   = "hbase-test-emr-key"
+  public_key = tls_private_key.emr_private_key.public_key_openssh
+  tags       = var.tags
+}
+
+# EMR Static HBase cluster
+module "emr-hbase" {
+  # source = "git::git@github.com:Datatamer/terraform-aws-emr.git?ref=6.2.0"
```
[the remark this answers is not in the document]
Comment (fvazquez-caylent) [reply examples/static-hbase-logs/emr.tf:20]: Fixed.

File: examples/static-hbase-logs/files/cloudwatch-install.tpl [hunk 3]
```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+# Grab local variables
+os=$(awk -F= '/^NAME/{print $2}' /etc/os-release)
+arch="amd64"
+id=$(TOKEN=`curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600"` \
```
[the remark this answers is not in the document]
Comment (fvazquez-caylent) [reply examples/static-hbase-logs/files/cloudwatch-install.tpl:5]: Good suggestion. Fixed.

File: examples/static-hbase-logs/networking.tf [hunk 4]
```diff
@@ -0,0 +1,40 @@
+module "sg-ports" {
+  # source               = "git::https://github.com/Datatamer/terraform-aws-emr.git//modules/aws-emr-ports?ref=6.2.0"
```
[the remark this answers is not in the document]
Comment (fvazquez-caylent) [reply examples/static-hbase-logs/networking.tf:2]: Fixed.

File: examples/static-hbase-logs/terraform.tfvars [hunk 5]
```diff
@@ -0,0 +1,9 @@
+bucket_name_for_root_directory = "poc-cw-root-directory-bucket"
+bucket_name_for_logs           = "poc-cw-logs-bucket"
+ingress_cidr_blocks            = []
+name_prefix                    = "poc-cw-without-vpc"
+key_pair                       = "fd-emr-test"
+vpc_id                         = "" # <---- Replace me
+application_subnet_id          = "" # <---- Replace me
+compute_subnet_id              = "" # <---- Replace me
+vpce_logs_endpoint_dnsname     = "" # <---- Replace me
```
[the remark this answers is not in the document]
Comment (fvazquez-caylent) [reply examples/static-hbase-logs/terraform.tfvars:9]: Deleted.

File: examples/static-spark-logs/emr.tf [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  source     = "terraform-aws-modules/key-pair/aws"
+  version    = "1.0.0"
+  key_name   = "spark-test-emr-key"
+  public_key = tls_private_key.emr_private_key.public_key_openssh
+  tags       = var.tags
+}
+
+# EMR Static Spark cluster
+module "emr-spark" {
+  # source = "git::git@github.com:Datatamer/terraform-aws-emr.git?ref=6.2.0"
```
[the remark this answers is not in the document]
Comment (fvazquez-caylent) [reply examples/static-spark-logs/emr.tf:20]: Fixed.

File: examples/static-spark-logs/networking.tf [hunk 7]
```diff
@@ -0,0 +1,38 @@
+module "sg-ports" {
+  # source  = "git::https://github.com/Datatamer/terraform-aws-emr.git//modules/aws-emr-ports?ref=6.2.0"
```
[the remark this answers is not in the document]
Comment (fvazquez-caylent) [reply examples/static-spark-logs/networking.tf:2]: Fixed.

File: main.tf [hunk 8]
```diff
@@ -1,6 +1,7 @@
 locals {
-  applications   = [for app in var.applications : lower(app)]
-  effective_tags = length(var.tags) > 0 ? var.tags : var.additional_tags
+  applications          = [for app in var.applications : lower(app)]
+  effective_tags        = length(var.tags) > 0 ? var.tags : var.additional_tags
```
[the remark this answers is not in the document]
Comment (fvazquez-caylent) [reply main.tf:3]: We were told to align on what we did with tags in S3.

File: modules/aws-emr-iam/main.tf (hunk 1 above)
Comment (fvazquez-caylent) [reply modules/aws-emr-iam/main.tf:723]: you are righ in here.

File: examples/static-spark-logs/terraform.tfvars [hunk 9]
```diff
@@ -0,0 +1,9 @@
+bucket_name_for_root_directory = "poc-cw-root-directory-bucket"
+bucket_name_for_logs           = "poc-cw-logs-bucket"
+ingress_cidr_blocks            = [] # Replace me
+name_prefix                    = "poc-cw-without-vpc"
+key_pair                       = "fd-emr-test"
+vpc_id                         = "vpc-01b63eaa58296f88c"                                             # <---- Replace me
+application_subnet_id          = "subnet-05d261107e6434fb1"                                          # <---- Replace me
+compute_subnet_id              = "subnet-09b99247bb12f7f39"                                          # <---- Replace me
+vpce_logs_endpoint_dnsname     = "vpce-0ef8d5634d7795af8-p0888dsr.logs.us-east-2.vpce.amazonaws.com" # <---- Replace me
```
[the remark this answers is not in the document]
Comment (fvazquez-caylent) [reply examples/static-spark-logs/terraform.tfvars:9]: Deleted.

File: modules/aws-emr-iam/main.tf (hunk 1 above)
Comment (fvazquez-caylent) [reply modules/aws-emr-iam/main.tf:723]: We replicated what we got in the S3 and Tamr VM modules.

**Comment by fvazquez-caylent** [comment]
Another one is opened for this.

**Pull request closed without merging**